[Website] Deploy step on deploy GitHub action has been failing for the last commits

The deploy step for the Arrow site is failing. After merging the changes are not being deployed at the moment:

See: https://github.com/apache/arrow-site/actions/workflows/deploy.yml
and a specific job: https://github.com/apache/arrow-site/actions/runs/8330658484/job/22795808856#step:9:52

```
 From https://github.com/apache/arrow-site
 * [new branch]              add-bryce  -> deploy/add-bryce
 * [new branch]              asf-site   -> deploy/asf-site
 * [new branch]              main       -> deploy/main
Switched to a new branch 'asf-site'
branch 'asf-site' set up to track 'deploy/asf-site'.
rsync: [sender] change_dir "/home/runner/work/arrow-site/arrow-site/../build/docs" failed: No such file or directory (2)
rsync error: some files/attrs were not transferred (see previous errors) (code 23) at main.c(1338) [sender=3.2.7]
Error: Process completed with exit code 23.
```

--- a/.github/workflows/deploy.yml
+++ b/.github/workflows/deploy.yml
@@ -103,10 +103,6 @@ jobs:
             --exclude '/docs/' \
             ../build/ \
             ./
-          rsync \
-            -a \
-            ../build/docs/ \
-            docs/
           touch .nojekyll
           if [ "$(git status --porcelain)" != "" ]; then
             # There are changes to the built site

